modules/virt-measuring-latency-vm-secondary-network.adoc

```diff
@@ -0,0 +1,229 @@
+
+.Prerequisites
+
+* You installed the OpenShift CLI (`oc`).
+* You logged in to the cluster as a user with the `cluster-admin` role.
+* The cluster has at least two worker nodes.
+* The Multus Container Network Interface (CNI) plug-in is installed on the cluster.
+* You configured a Linux bridge network attachment definition for a namespace.
+
+.Procedure
+
+. Create a configuration file that contains the resources to set up the framework. This includes a namespace and service account for the framework, and the `ClusterRole` and `ClusterRoleBinding` objects to define permissions for the service account.
++
+.Example framework manifest file
+[%collapsible]
+====
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: k8s-checkup-framework
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: checkup-framework-sa
+  namespace: k8s-checkup-framework
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: cluster-admin-to-checkup-framework-sa
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: checkup-framework-sa
+  namespace: k8s-checkup-framework
+----
+====
+
+. Apply the framework manifest:
++
+[source,terminal]
+----
+$ oc apply -f <framework-manifest>.yaml
+----
+
+. Create a configuration file that contains the `ClusterRole` and `Role` objects with permissions that are required by the checkup to access the cluster:
++
+.Example checkup role manifest file
+[%collapsible]
+====
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: nodes-reader
+rules:
+- apiGroups: [""]
+  resources: ["nodes"]
+  verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kubevirts-reader
+rules:
+- apiGroups: ["kubevirt.io"]
+  resources:
+  - kubevirts
+  verbs:
+  - get
+  - list
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: networks-reader
+rules:
+- apiGroups: ["k8s.cni.cncf.io"]
+  resources: ["network-attachment-definitions"]
+  verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kubevirt-vmis-manager
+rules:
+- apiGroups: ["kubevirt.io"]
+  resources: ["virtualmachineinstances"]
+  verbs:
+  - get
+  - create
+  - delete
+- apiGroups: ["subresources.kubevirt.io"]
+  resources: ["virtualmachineinstances/console"]
+  verbs: ["get"]
+----
+====
+
+. Apply the checkup roles manifest:
++
+[source,terminal]
+----
+$ oc apply -f <latency-roles>.yaml
+----
+
+. Create a `ConfigMap` manifest that contains the environment variables for the checkup. The config map provides input for the framework to run the checkup and also stores the results of the checkup.
++
+.Example input config map
+[source,yaml]
+----
+apiVersion: v1
```

Comment:
Can you please sync with the one we use in the tests?
https://github.com/kiagnose/kiagnose/blob/fd0aa6851235243278eaadf9d168fd30ebd521f0/checkups/kubevirt-vm-latency/automation/e2e.sh#L170-L186

Maybe it will be better all over to use the `cat <<EOF | oc apply -f -` pattern, then this will be just a copy-paste for the reader.

Side note: all the `${something}` are variables, which probably should be replaced to be explicit. Also the `KUBECTL` on is in D/S case `oc`.

I hope I did not complicated too much by ref a script. If needed, I can make an explicit suggestion.